components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java

```diff
@@ -735,9 +740,13 @@ private void writeToFile(URI parentName, String newFileName, String content) thr
             handleException("Unable to create parent directory: " + parentName);
         }
         File newFile = new File(parent, newFileName);
-        try (BufferedWriter writer = new BufferedWriter(new FileWriter(newFile))) {
+        File newMetadataFile = new File(parent, newFileName + METADATA_FILE_SUFFIX);
+
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(newFile));
+             BufferedWriter metadataWriter = new BufferedWriter(new FileWriter(newMetadataFile))) {
             writer.write(content);
             writer.flush();
+            metadata.store(metadataWriter, null);
```

Comment:
Rather passing `null` into a method, shall we overload the method?

Reply:
Its a method given by Java API. No such overloaded method available.